core/modules/bpf.cc

```diff
@@ -125,6 +127,25 @@ CommandResponse BPF::CommandAdd(const bess::pb::BPFArg &arg) {
   return CommandSuccess();
 }
 
+CommandResponse BPF::CommandDelete(const bess::pb::BPFArg &arg) {
+  for (const auto &f : arg.filters()) {
+    if (f.gate() < 0 || f.gate() >= MAX_GATES) {
+      return CommandFailure(EINVAL, "Invalid gate");
+    }
+
+    for (auto i = filters_.begin(); i != filters_.end(); ++i) {
+        if (f.priority() == i->priority &&
+            f.gate() ==i->gate &&
+            f.filter() == i->exp) {
+            filters_.erase(i);
+            i--;
+        }
+    }
```

Comment:
I don't think `i` is guaranteed to be valid after calling `filters_.erase()` perhaps use this instead?
 
```suggestion
    for (auto i = filters_.begin(); i != filters_.end();) {
        if (f.priority() == i->priority &&
            f.gate() ==i->gate &&
            f.filter() == i->exp) {
            i = filters_.erase(i);
        } else {
            i++;
        }
    }
```

Reply:
Oh, thank you. It actually needs to break the loop :)